dali/operators/reader/loader/fits_loader.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dirent.h>
+#include <errno.h>
+#include <cstdlib>
+#include <memory>
+
+#include <fitsio.h>
+#include "dali/core/common.h"
+#include "dali/operators/reader/loader/fits_loader.h"
+#include "dali/operators/reader/loader/utils.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/util/file.h"
+#include "dali/util/fits.h"
+
+#define max_number_of_axes 999
+
+namespace dali {
+
+void FitsLoader::ReadSample(FitsFileWrapper& target) {
+  auto filename = files_[current_index_];
+  fitsfile* current_file;
+  int status = 0, hdunum;
+  Index pastIndex = current_index_;
+
+  // hanlde moving through HDUs, assuming that primary hdu is for meta data
+  auto path = filesystem::join_path(file_root_, filename);
+  fits_open_file(&current_file, path.c_str(), READONLY, &status);
+  fits_get_num_hdus(current_file, &hdunum, &status);
+  fits_movabs_hdu(current_file, hdu_index_, NULL, &status);
+
+  if (hdu_index_ < hdunum) {
+    hdu_index_++;
```

Comment:
The suggested approach would mean, that we potentially read one sample consisting of multiple HDUs.